boards/arm/raspberrypi_pico/raspberrypi_pico-pinctrl.dtsi

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2021, Yonatan Schachter
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/pinctrl/rp2040-pinctrl.h>
+
+&pinctrl {
+	uart0_default: uart0_default {
+		pinmux = <UART0_TX_P0>, <UART0_RX_P1>;
+	};
+};
```

Comment:
this mixes grouping and per-pin node strategies. You need to add grouping capability to achieve this.

You now have two implementations that do this (adding PRs so that you can see all pieces):
https://github.com/zephyrproject-rtos/zephyr/pull/39429
https://github.com/zephyrproject-rtos/zephyr/pull/39974

And a documented example:
https://docs.zephyrproject.org/latest/guides/pinctrl/index.html#pin-configuration